Analysis/Tasks/PWGHF/taskX.cxx

```diff
@@ -0,0 +1,107 @@
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_prong2;
+using namespace o2::framework::expressions;
+
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  ConfigParamSpec optionDoMC{"doMC", VariantType::Bool, false, {"Fill MC histograms."}};
+  workflowOptions.push_back(optionDoMC);
+}
+
+#include "Framework/runDataProcessing.h"
+
+namespace o2::aod
+{
+namespace extra
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);
+}
+DECLARE_SOA_TABLE(Colls, "AOD", "COLLSID", o2::aod::extra::CollisionId);
+} // namespace o2::aod
+struct AddCollisionId {
+  Produces<o2::aod::Colls> colls;
+  void process(aod::HfCandProng2 const& candidates, aod::Tracks const&)
+  {
+    for (auto& candidate : candidates) {
+      colls(candidate.index0_as<aod::Tracks>().collisionId());
+    }
+  }
+};
+
+/// X(3872) analysis task
+/// FIXME: Still need to remove track duplication!!!
+struct TaskX {
+  HistogramRegistry registry{
+    "registry",
+    {{"hMassJpsi", "2-prong candidates;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH1F, {{500, 0., 5.}}}},
+     {"hPtCand", "X candidates;candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}}}};
+
+  Configurable<int> selectionFlagJpsi{"selectionFlagJpsi", 1, "Selection Flag for Jpsi"};
+  Configurable<double> cutEtaCandMax{"cutEtaCandMax", -1., "max. cand. pseudorapidity"};
+
+  Filter filterSelectCandidates = (aod::hf_selcandidate_jpsi::isSelJpsiToEE >= selectionFlagJpsi);
+
+  /// aod::BigTracks is not soa::Filtered, should be added when filters are added
+  void process(aod::Collision const&, aod::BigTracks const& tracks, soa::Filtered<soa::Join<aod::HfCandProng2, aod::HFSelJpsiToEECandidate, aod::Colls>> const& candidates)
+  {
+    for (auto& candidate : candidates) {
+      if (!(candidate.hfflag() & 1 << JpsiToEE)) {
+        continue;
+      }
+      if (cutEtaCandMax >= 0. && std::abs(candidate.eta()) > cutEtaCandMax) {
+        continue;
+      }
+      registry.fill(HIST("hMassJpsi"), InvMassJpsiToEE(candidate));
+
+      int index0jpsi = candidate.index0Id();
+      int index1jpsi = candidate.index1Id();
+      for (auto& trackPos : tracks) {
+        if (trackPos.signed1Pt() < 0) {
+          continue;
+        }
+        if (trackPos.globalIndex() == index0jpsi) {
+          continue;
+        }
+        for (auto& trackNeg : tracks) {
+          if (trackNeg.signed1Pt() > 0) {
+            continue;
+          }
+          if (trackNeg.globalIndex() == index1jpsi) {
+            continue;
+          }
+          registry.fill(HIST("hPtCand"), candidate.pt() + trackPos.pt() + trackNeg.pt());
+        } // pi- loop
+      }   // pi+ loop
+    }     // Jpsi loop
+  }       // process
+};        // struct
+
+WorkflowSpec defineDataProcessing(ConfigContext const&)
+{
+  WorkflowSpec workflow{
+    adaptAnalysisTask<AddCollisionId>("hf-task-add-collisionId"),
```

Comment:
It would be better to give this task a unique name because there is already an identical one in `taskBplus.cxx`.
I would suggest `hf-task-x-add-collisionId`.

Reply:
changed it as you suggested